### PR TITLE
Semantic tokens encoder

### PR DESCRIPTION
**Pull request by keynmol**
TODO:
- Nothing, honestly this is pretty much perfect
- Test corner cases

### CODE REVIEW

File: modules/lsp/src/main/scala/tools/SemanticTokensEncoder.scala [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+):
+  val legend = SemanticTokensLegend(
+    tokenTypes = tokensIndex.keys.map(_.raw).toVector,
+    tokenModifiers = modifiersIndex.keys.map(_.raw).toVector
+  )
+
+  def encode(v: Vector[SemanticToken]) =
+    import types.*
+    val sorted =
+      v.sortBy(tok => (tok.position.line.value, tok.position.character.value))
```
Comment (keynmol) [inline modules/lsp/src/main/scala/tools/SemanticTokensEncoder.scala:23]: I wonder if it's strictly necessary?
